packages/tables/backend/sirius-components-collaborative-tables/src/main/java/org/eclipse/sirius/components/collaborative/tables/handlers/EditTextfieldCellEventHandler.java

```diff
@@ -0,0 +1,155 @@
+        this.counter = Counter.builder(Monitoring.EVENT_HANDLER)
+                .tag(Monitoring.NAME, this.getClass().getSimpleName())
+                .register(meterRegistry);
+        // @formatter:on
+    }
+
+    @Override
+    public boolean canHandle(ITableInput tableInput) {
+        return tableInput instanceof EditTextfieldCellInput;
+    }
+
+    @Override
+    public void handle(One<IPayload> payloadSink, Many<ChangeDescription> changeDescriptionSink, IEditingContext editingContext, Table table, TableDescription tableDescription, ITableInput tableInput) {
+        this.counter.increment();
+
+        ChangeDescription changeDescription = new ChangeDescription(ChangeKind.NOTHING, tableInput.representationId(), tableInput);
+        String message = this.messageService.invalidInput(tableInput.getClass().getSimpleName(), EditTextfieldCellInput.class.getSimpleName());
+        IPayload payload = new ErrorPayload(tableInput.id(), message);
+
+        if (tableInput instanceof EditTextfieldCellInput) {
+            EditTextfieldCellInput input = (EditTextfieldCellInput) tableInput;
+            var optCell = this.tableQueryService.findCellById(table, UUID.fromString(input.cellId()));
+
+            if (optCell.isPresent()) {
+                AbstractCell cell = optCell.get();
+                var optLine = this.tableQueryService.findLineById(table, cell.getParentLineId());
+                var optCol = this.tableQueryService.findColumnById(table, cell.getColumnId());
+
+                if (optLine.isPresent() && optCol.isPresent()) {
+                    this.invokeEditCell(cell, optLine.get(), optCol.get(), editingContext, tableDescription, input.newValue());
+
+                    changeDescription = new ChangeDescription(ChangeKind.SEMANTIC_CHANGE, input.representationId(), input);
+                    payload = getPayload(input.id());
+                }
+            }
+        }
+
+        payloadSink.tryEmitValue(payload);
+        changeDescriptionSink.tryEmitNext(changeDescription);
+    }
+
+    private void invokeEditCell(AbstractCell cell, Line line, Column column, IEditingContext editingContext, TableDescription tableDescription, String newValue) {
+        var optionalLineDescription = this.tableDescriptionService.findLineDescriptionById(tableDescription, line.getDescriptionId());
+
+        if (optionalLineDescription.isPresent()) {
+            LineDescription lineDescription = optionalLineDescription.get();
+
+            var optionalSelf = this.objectService.getObject(editingContext, line.getTargetObjectId());
+            if (optionalSelf.isPresent()) {
+                Object self = optionalSelf.get();
+
+                VariableManager variableManager = new VariableManager();
+                variableManager.put(VariableManager.SELF, self);
+                variableManager.put(IEditingContext.EDITING_CONTEXT, editingContext);
+                variableManager.put("featureName", column.getFeatureName()); //$NON-NLS-1$
```

Comment:
As another example, I could have for my columns some `EObjects` computed using `self.eAllContents(papaya::Component)` and for my lines some other `EObjects` but this time some `EAttributes` `papaya::Component.eAllReferences` and for each cell something like `aql:columnSemanticElement.eGet(lineSemanticElement)->first()`.

This would only display the first semantic element of any reference from a component. Each cell would thus be an EObject with a label computed with a Java service using EMF Edit LabelProvider `aql:self.getLabel()`.